### DISCUSSION TITLE
Can this software render a video by going to export -> render video?

### DISCUSSION THREAD

**Discussion by JarrodAI** [joonaspaakko/Batch-Mockup-Smart-Object-Replacement-photoshop-script#28] [Q&A]
I have a .psd video mockup I'd like to do with this. Of course I am new and would need to get the actual software running and tested first but good things to think about. :)  Thank you for this!

**Comment by joonaspaakko** [comment]
Unfortunately no. In my defense, I dont think I've ever used one. At least in my life Photoshop video mockups have been incredibly rare. I've always used After Effects for video mockups and never in batch.

**Reply by bajwaprabh** [reply]
sorry for asking again as you already explained well, but wanna know if it's possible to do with some customizations in code or not?

**Reply by joonaspaakko** [reply]
Sure it could be modified to do that. The biggest hangups might be:

1. I'm not sure if it would be capable importing video files as is
    - The script method for importing input files might not be compatible with video files. I'm not sure.
    - If it is compatible, just changing the input file format could do...
2. It doesn't understand timelines
    - I think you may need to enable timeline after importing input files.
    - Could potentially be done with the input file placed script callback or action.
3. It doesn't have a way to export as a video format

**Reply by bajwaprabh** [reply]
Almost did everything but stuck at 3rd point don't know how to export it as a video.

Even I tried changing the file output name to mp4 and other video formats but not working
var output = {
  path: '../$/ sstt(output)', 
  format: 'jpg', ------------------------------- here 
  folders: true,
  filename: '@input - $',
};

**Reply by joonaspaakko** [reply]
There's nothing in the script that explicitly handles videos in any way. Actually, I've never scripted a video export in PS, but I would imagine it's similar to any image export, except maybe it uses a different method entirely. 

When you set the output path as `"psd"` for example, it takes output options from line `525`

https://github.com/joonaspaakko/Batch-Mockup-Smart-Object-Replacement-photoshop-script/blob/master/script/Batch%20Mockup%20Smart%20Object%20Replacement.jsx#L523-L534

And feeds them to the `saveAs()` method on line `270`:

https://github.com/joonaspaakko/Batch-Mockup-Smart-Object-Replacement-photoshop-script/blob/fc546baacf6ea29cab95eb3b97cb869ae2baeccd/script/Batch%20Mockup%20Smart%20Object%20Replacement.jsx#L270

**Reply by bajwaprabh** [reply]
function 
export2(directory, name2, ameFormatName, amePresetName, useDocumentSize, frameRate, manage, selectedFrames, quality, Z3DPrefHighQualityErrorThreshold) 
{
	var s2t = function (s) {
		return app.stringIDToTypeID(s);
		
		
this export function is found in Adobe community forms, it will work if we create a new video output calling function?

**Reply by bajwaprabh** [reply]
(https://community.adobe.com/t5/photoshop-ecosystem-discussions/how-to-render-a-mp4-animation-from-jsx-script/m-p/11848198)

I'm trying the Adobe Scriptlistener plug-in but it's very confusing, it possible for you to create a function or modify the above so that it will work in your code?

**Comment by kimle0578** [comment]
Hey @bajwaprabh, I'm also looking for the same. Are you able to get video as output?
